src/Core/Core.idr

```diff
@@ -449,10 +449,10 @@ coreFail e = MkCore (pure (Left e))
 export
 wrapError : (Error -> Error) -> Core a -> Core a
 wrapError fe (MkCore prog)
-    = MkCore (prog >>=
-                 (\x => pure $ case x of
-                             Left err => Left (fe err)
-                             Right val => Right val))
+    = MkCore $ prog >>=
+                    pure . \case
+                      Left err => Left $ fe err
+                      Right val => Right val
```

Comment:
As suggested earlier, this is a good opportunity to use `mapFst` instead.

Reply:
added